### ISSUE TITLE
Several warnings when building

### ISSUE THREAD

**Issue by eribertomota** [resurrecting-open-source-projects/nbtscan#1]
The warnings are identical.

```
nbtscan.c:123:5: warning: $ operand number used after format without operand number [-Wformat=]
  123 |     printf("Transaction ID: 0x%04x (%1$d)\n", hostinfo->header->transaction_id);
      |     ^~~~~~
```

**Comment by codenstuff** [comment]
This warning shows up due to mixing conversion specifier styles. The style using '$' is not part of the C99 standard and is meant to be used throughout your code for all conversions. The reason it was used in this case, is because the '$' argument allows you to point to a specific argument included in the list of arguments for the printf() statement. This allows for multiple conversion specifiers in a single printf() call to point to the same argument.

```C
printf("Transaction ID: 0x%04x (%1$d)\n", hostinfo->header->transaction_id);
```

The above line is using **hostinfo->header->transaction_id** as the argument for both %04x and %1$d. One way I've tested that works to get rid of the warnings is to replace those calls with appropriate macros included in the <inttypes.h> library. The previous code would then look like this:
```C
printf("Transaction ID: 0x%04x (%" PRIu16 ")\n", hostinfo->header->transaction_id, hostinfo->header->transaction_id);
```
PRIu16 is just a macro that translates to the appropriate specifier for a 16-bit unsigned int at runtime. This makes the code more portable when used on both 32-bit and 64-bit, because the format specifiers used in printf() are not the same for fixed-width integer types.

Let me know if you think this is a sound solution to the warnings and I can make a pull request.

**Comment by eribertomota** [comment]
Hi @codenstuff,

Thanks a lot for your help and ideas. Note that I am not a C programmer. So, yes!, a PR is very welcome. However, I will add you to project to allow you commit directly in source code. Please, add your name in copyright lines in headers.

Feel free to do more changes and fixes.

Cheers,

Eriberto

**Comment by eribertomota** [comment]
Hi @codenstuff,

Do you still interested to send a pull request? Thanks!

**Comment by eribertomota** [comment]
Closed via #3
